src/relay_protocol.rs

```diff
@@ -70,6 +82,27 @@ impl RelayResponseCode {
     }
 }
 
+#[derive(Debug, Copy, Clone)]
+pub enum MessageCode {
+    Op(RelayOpCode),
+    Response(RelayResponseCode),
+}
+
+impl MessageCode {
+    pub fn from_bytes(bytes: [u8; 4], context_id: MessageContext, is_client: bool) -> Option<Self> {
+        Some(match context_id.client_initiated() == is_client {
+            true => MessageCode::Response(RelayResponseCode::from_bytes(bytes)),
+            false => MessageCode::Op(RelayOpCode::from_bytes(bytes)?),
+        })
+    }
+    pub fn to_bytes(self) -> [u8; 4] {
+        match self {
+            MessageCode::Op(op_code) => op_code.to_bytes(),
+            MessageCode::Response(resp_code) => resp_code.to_bytes(),
+        }
+    }
+}
```

Comment:
I'm not quite sure why we need this. Shouldn't the direction of the message be apparent based on if it was written by the client code or read off of the wire? Although I see that this the context ID exists in the protocol.

Reply:
Thinking about this more I think this was supposed to be a request ID. We just set it to zero for now because we didn't have a use. So i think we should leave that rather than having this be server vs client (which should be known based on the direction of the message).

Reply:
What should the `client_initiated(id)` helper you suggested do? I thought you want it to check if the id belongs to a client generated exchange, which is exactly what we have in the PR.

What would you generate with the id as input? If you meant a random id generator, then I would delay that until we actually use generated ids in any codebase.

Reply:
Because it is an unused filler value right now. It just needs to fit the protocol.

Reply:
Sounds good.